### PR TITLE
Allow Dockerfile content to come from stdin

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Fixes https://github.com/projectatomic/buildah/issues/886

**Comment by vrothberg** [comment]
Code LGTM.

@rhatdan Could you add an integration test for that?

**Comment by TomSweeneyRedHat** [comment]
@pixdrift in case you didn't see this one.

**Comment by TomSweeneyRedHat** [comment]
First blush LGTM, test would be good, but want to run a few tests before LGTM'ing

**Comment by vrothberg** [comment]
:+1: Thanks for the tests. LGTM!

File: imagebuildah/build.go [hunk 1]
```diff
@@ -1199,7 +1199,7 @@ func BuildDockerfiles(ctx context.Context, store storage.Store, options BuildOpt
 				contents.Close()
 				return errors.Wrapf(err, "error reading info about %q", dfile)
 			}
-			if dinfo.Size() == 0 {
+			if dinfo.Mode().IsRegular() && dinfo.Size() == 0 {
```
Comment (TomSweeneyRedHat) [inline imagebuildah/build.go]: Is this needed for this change, or just an unrelated tweak?

Comment (rhatdan) [reply imagebuildah/build.go]: Needed, this is why your tests with -f /dev/stdin were not working.
Size of a pipe is 0, If we only check on size of files, then the pipe will go into the imagebuilder and the build succeeds.

**Comment by TomSweeneyRedHat** [comment]
Red Hat CI is throwing up rhatdan:

```
make[1]: Leaving directory '/root/gopath/src/github.com/projectatomic/buildah/docs'
cmd/buildah/bud.go:31::warning: cyclomatic complexity 42 of function budCmd() is high (> 40) (gocyclo)
```

It's always easy....

**Comment by TomSweeneyRedHat** [comment]
Once the tests are happy, LGTM.   Works fine in my testing.

File: imagebuildah/build.go (hunk 1 above)
Comment (TomSweeneyRedHat) [reply imagebuildah/build.go]: thx for the 411

**Comment by TomSweeneyRedHat** [comment]
gofmt isn't your friend @rhatdan 

```
Error: source files are not formatted according to recommendations. Run "gofmt -s -w" on:
./imagebuildah/build.go
make: *** [validate] Error 1
```

**Comment by pixdrift** [comment]
Thanks @tomsweeney for raising the issue, it was on today's todo list! Looks like a neat solution @rhatdan and I will be able to quickly make use of it, so very much appreciated!

**Comment by TomSweeneyRedHat** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**